include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
 #include "network_interface.h"
 #include "timer_interface.h"
 
+/**
+ * @functionspage{mqtt,MQTT library}
+ *
+ * API functions
+ * - @functionname{mqtt_function_init}
+ * - @functionname{mqtt_function_free}
+ * - @functionname{mqtt_function_connect}
+ * - @functionname{mqtt_function_publish}
+ * - @functionname{mqtt_function_subscribe}
+ * - @functionname{mqtt_function_resubscribe}
+ * - @functionname{mqtt_function_unsubscribe}
+ * - @functionname{mqtt_function_disconnect}
+ * - @functionname{mqtt_function_yield}
+ * - @functionname{mqtt_function_attempt_reconnect}
+ * - @functionname{mqtt_function_get_next_packet_id}
```

Comment:
Why are those functions not declared in this file have their doxygen tags mentioned in this file (instead of `iot_mqtt_client.h`)?

Reply:
To keep the functions under the same page for doxygen grouping